components/Button/Button.jsx

```diff
@@ -1,91 +1,329 @@
-
-const iconSize = ({ size }) => {
-  const sizes = {
-    xsmall: '16px',
-    small: '16px',
-    normal: '24px',
-    large: '24px',
-    xlarge: '24px',
-  };
-
-  return `font-size: ${sizes[size] || sizes.normal};`;
-};
 
 const ButtonIcon = styled(Icon)`
   pointer-events: none;
 `;
 
 const StyledButton = styled.button`
-  ${buttonStyle}
+  align-items: center;
+  display: flex;
+  font-weight: bold;
+  justify-content: center;
+  letter-spacing: 0.2px;
+  border-radius: 4px;
+
+  ${props => `cursor: ${props.disabled ? 'not-allowed' : 'pointer'};`}
+
+  ${({ size, theme: { baseFontSize } }) => {
+    const sizes = {
+      xsmall: `${baseFontSize * 0.75}px`,
+      small: `${baseFontSize * 0.75}px`,
+      medium: `${baseFontSize}px`,
+      large: `${baseFontSize * 1.25}px`,
+      xlarge: `${baseFontSize * 1.5}px`,
+    };
+    return `font-size: ${sizes[size] || sizes.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { large, xlarge, xxlarge, xxxlarge },
+    },
+  }) => {
+    const heights = {
+      xsmall: `${large}px`,
+      small: `${xlarge}px`,
+      medium: `${xxlarge}px`,
+      large: `${xxxlarge}px`,
+    };
+
+    return `height: ${heights[size] || heights.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { small, medium },
+    },
+  }) => {
+    const paddings = {
+      xsmall: `0 ${small}px`,
+      small: `0 ${small}px`,
+      medium: `0 ${medium}px`,
+      large: `0 ${medium}px`,
+      xlarge: `0 ${medium}px`,
+    };
+
+    return `padding: ${paddings[size] || paddings.medium};`;
+  }}
+
+  ${props =>
+    props.center &&
+    `
+    margin-left: auto;
+    margin-right: auto;
+  `}
 
   ${ButtonIcon} {
-    ${iconSize}
+    margin-right: 5px;
   }
+
+  transition: all 0.2s ease-in-out;
+
+  ${props => props.full && `width: 100%;`}
+
+  ${({ skin, theme, disabled, stroked }) => {
+    const {
+      components: {
+        button: {
+          skins: {
+            [skin]: {
+              mainColor: {
+                100: mainColor100,
+                500: mainColor500,
+                700: mainColor700,
+              },
+              text: { 100: text100 },
+            },
+          },
+        },
+      },
+      colors: {
+        neutral: { 100: neutral100, 500: neutral500 },
+      },
+    } = theme;
+
+    let bgColor;
+    let textColor;
+
+    if (disabled && stroked) {
+      bgColor = neutral100;
+      textColor = neutral500;
+    } else if (stroked) {
+      bgColor = neutral100;
+      textColor = mainColor500;
+    } else if (disabled) {
+      bgColor = neutral500;
+      textColor = neutral100;
+    } else {
+      bgColor = mainColor500;
+      textColor = text100;
+    }
+
+    return `
+      background-color: ${bgColor};
+      color: ${textColor};
+
+      border: 1.5px solid ${disabled ? neutral500 : mainColor500};
+
+      ${shadow(2, neutral500)({ theme })}
+
+      :hover {
+        ${
+          !disabled
```

Comment:
What do you guys think of using an logical and (&&) here instead of a ternary, since has no else condition?
(in `focus` and `active` too)

Reply:
Using `&&` will return a boolean condition what will make a string template to be like this:
 ```
   :hover { false; }
```
We use a ternary expression to prevent that situation.